app/javascript/components/util/DeleteDialog.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Button, Form, Modal } from 'react-bootstrap';
+
+class DeleteDialog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      disabled: true,
+      showTextInput: false,
+    };
+  }
+
+  handleReasonChange = event => {
+    event.persist();
+    const showTextInput = event.target.value === 'Other';
+    this.setState({ disabled: false, showTextInput, [event.target.id]: event.target.value }, () => {
+      this.props.onChange(event);
+    });
+  };
+
+  delete = () => {
+    this.setState({ loading: true }, () => {
+      this.props.delete();
+    });
+  };
+
+  render() {
+    return (
+      <Modal size="lg" show centered onHide={this.props.toggle}>
+        <Modal.Header>
+          <Modal.Title>Delete {this.props.type}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <p>
+            Are you sure you want to delete this {this.props.type}? This action cannot be undone. For auditing purposes, this deletion will be available in this
+            record&apos;s history export.
+          </p>
+          <p>Please select reason for deletion:</p>
+          <Form.Control as="select" className="form-control-md mb-3" id="delete_reason" onChange={this.handleReasonChange} defaultValue={-1}>
+            <option disabled value={-1}>
+              --
+            </option>
+            <option>Duplicate entry</option>
+            <option>Entered in error</option>
+            <option>Other</option>
+          </Form.Control>
+          {this.state.showTextInput && (
+            <Form.Control
```

Comment:
let's put a character limit on this and make sure that's validated in the back end as well

![Uploading Screen Shot 2021-05-05 at 2.54.17 AM.png…]()

Reply:
@holmesie for the back end validation, would we rather add a character limit validation rule to the model or add an inline if check with `redirect_to root_url && return`